### PR TITLE
gpt: add search by disk uuid command

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

**Comment by marineam** [comment]
Please match the formatting of the existing code, arbitrary white space changes to code from upstream is just going to lead to merge conflicts later on.

**Comment by crawford** [comment]
:( There were tabs randomly sprinkled throughout the code.

I'll re-sprinkle.

**Comment by marineam** [comment]
GNU uses 8 space tabs, so they are in fact supposed to be there. From my vimrc:

```
" GNU
"au FileType c   setlocal noexpandtab tabstop=8 softtabstop=2 shiftwidth=2
"au FileType cpp setlocal noexpandtab tabstop=8 softtabstop=2 shiftwidth=2
```

**Comment by crawford** [comment]
Updated.

File: grub-core/commands/search.c [hunk 1]
```diff
@@ -131,6 +132,25 @@ iterate_device (const char *name, void *data)
...
+    {
+      grub_device_t dev;
+      char *quid;
+
+      dev = grub_device_open (name);
+      if (dev)
+        {
+          if (grub_gpt_disk_uuid (dev, &quid) == GRUB_ERR_NONE)
+            {
+              if (grub_strcmp (quid, ctx->key) == 0)
```
Comment (marineam) [inline grub-core/commands/search.c]: should have some tabs

File: tests/gpt_unit_test.c [hunk 2]
```diff
@@ -614,6 +614,37 @@ search_uuid_test (void)
...
+  test_result = grub_env_get ("test_result");
+  grub_test_assert (test_result && strcmp (test_result, expected_result) == 0,
+                    "wrong device: %s (%s)", test_result, expected_result);
+  grub_free (expected_result);
+
+  grub_env_unset ("test_result");
+  grub_search_disk_uuid (bogus_uuid, "test_result", 0, NULL, 0);
+  test_result = grub_env_get ("test_result");
+  grub_test_assert (test_result == NULL,
+                    "unexpected device: %s", test_result);
```
Comment (marineam) [inline tests/gpt_unit_test.c]: tab

File: tests/gpt_unit_test.c [hunk 3]
```diff
@@ -614,6 +614,37 @@ search_uuid_test (void)
...
+  grub_test_assert (grub_errno == GRUB_ERR_FILE_NOT_FOUND,
+                    "unexpected error: %s", grub_errmsg);
```
Comment (marineam) [inline tests/gpt_unit_test.c]: tab

**Comment by marineam** [comment]
Almost! (yes the GNU style is a bit silly)

**Comment by crawford** [comment]
Tabbed. I actually like the consistency of GNU style. Too bad they use tabs incorrectly.

**Comment by marineam** [comment]
[LGTM](http://lgtm.in/i/S30pMumWp)

**Pull request merged**